edweiss-firebase/functions/src/functions/action/generateQuizContentFromMaterial.ts

```diff
@@ -0,0 +1,138 @@
+import prompt from 'actions/ai';
+import Quizzes from 'model/quizzes';
+import { onSanitizedCall } from 'utils/firebase';
+import { Collections, getRequiredDocument } from 'utils/firestore';
+import { Predicate } from 'utils/sanitizer';
+import { COURSE_NOT_FOUND, fail, ok } from 'utils/status';
+const pdf = require('pdf-parse');
+import admin = require('firebase-admin');
+
+export const generateQuizContentFromMaterial = onSanitizedCall(Quizzes.Functions.generateQuizContentFromMaterial, {
+	courseId: Predicate.isNonEmptyString,
+	materialUrl: Predicate.isNonEmptyString
+}, async (userId, args) => {
+	const course = await getRequiredDocument(Collections.courses, args.courseId, COURSE_NOT_FOUND);
+	const [rawData] = await admin.storage().bucket().file(args.materialUrl).download();
+
+	let pdfContent = "";
+	try {
+		pdfContent = (await pdf(rawData)).text;
+	} catch (e) {
+		return fail("error-parsing-pdf");
+	}
+
+	if (pdfContent == undefined || pdfContent == null)
+		return fail("pdf-content-undefined");
+
+	/**
+questionmcq<TOKEN_END_QUESTION>
```

Comment:
A nitpick: this comment can be removed.

Reply:
I think it should instead be formally explicited as the "question" format for potential future refactors.

Reply:
Ah, I see! Thanks for clearing up the confusion. UwU